commands/payment_channel_daemon_test.go

```diff
@@ -1,547 +1,598 @@
-	"github.com/filecoin-project/go-filecoin/fixtures"
-	th "github.com/filecoin-project/go-filecoin/testhelpers"
+	"github.com/filecoin-project/go-filecoin/tools/fast"
+	"github.com/filecoin-project/go-filecoin/tools/fast/fasting"
+	"github.com/filecoin-project/go-filecoin/tools/fast/series"
 	"github.com/filecoin-project/go-filecoin/types"
 )
 
+func init() {
+	logging.SetDebugLogging()
+	series.GlobalSleepDelay = 5 * time.Second
+}
+
+func VoucherFromString(data string) (paymentbroker.PaymentVoucher, error) {
+	_, cborVoucher, err := multibase.Decode(data)
+	if err != nil {
+		return paymentbroker.PaymentVoucher{}, err
+	}
+
+	var voucher paymentbroker.PaymentVoucher
+	if err = cbor.DecodeInto(cborVoucher, &voucher); err != nil {
+		return paymentbroker.PaymentVoucher{}, err
+	}
+
+	return voucher, nil
+}
+
 func TestPaymentChannelCreateSuccess(t *testing.T) {
 	t.Parallel()
-	assert := assert.New(t)
+	require := require.New(t)
 
-	d := makeTestDaemonWithMinerAndStart(t)
-	defer d.ShutdownSuccess()
-
-	args := []string{"paych", "create"}
-	args = append(args, "--from", fixtures.TestAddresses[0], "--gas-price", "0", "--gas-limit", "300")
-	args = append(args, fixtures.TestAddresses[1], "10000", "20")
-
-	paymentChannelCmd := d.RunSuccess(args...)
-	messageCid, err := cid.Parse(strings.Trim(paymentChannelCmd.ReadStdout(), "\n"))
-	require.NoError(t, err)
-
-	var wg sync.WaitGroup
-
-	wg.Add(1)
-	go func() {
-		wait := d.RunSuccess("message", "wait",
-			"--return",
-			"--message=false",
-			"--receipt=false",
-			messageCid.String(),
-		)
-		_, ok := types.NewChannelIDFromString(strings.Trim(wait.ReadStdout(), "\n"), 10)
-		assert.True(ok)
-		wg.Done()
+	// This test should run in 20 block times
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(20*series.GlobalSleepDelay))
+	defer cancel()
+
+	// Get basic testing environment
+	env, _, NewNode, ctx := fasting.BasicFastSetup(ctx, t, fast.EnvironmentOpts{})
+
+	// Teardown after test ends
+	defer func() {
+		err := env.Teardown(ctx)
+		require.NoError(err)
 	}()
 
-	d.RunSuccess("mining once")
+	// Start test
+	targetDaemon := NewNode()
+	payerDaemon := NewNode()
+
+	addrs, err := targetDaemon.AddressLs(ctx)
+	require.NoError(err)
+	targetAddr := addrs[0]
+
+	addrs, err = payerDaemon.AddressLs(ctx)
+	require.NoError(err)
+	payerAddr := addrs[0]
+
+	channelExpiry := types.NewBlockHeight(20)
+	channelAmount := types.NewAttoFILFromFIL(1000)
 
-	wg.Wait()
+	WithPaymentChannel(ctx, t, payerDaemon, payerAddr, targetAddr, channelAmount, channelExpiry, func(chanid *types.ChannelID) {
+		require.NotNil(chanid)
+	})
+}
+
+// WithPaymentChannel takes a filecoin daemon and constructs a payment channel to the `target` for the `amt` for `eol`, and calls fn with the channel id
+func WithPaymentChannel(ctx context.Context, t *testing.T, p *fast.Filecoin, from, target address.Address, amt *types.AttoFIL, eol *types.BlockHeight, fn func(chanid *types.ChannelID)) {
```

Comment:
Nit: move all the helper methods below all the test methods so that a top-down reading of the file describes what is being tested, then delves into progressive detail.